packages/profiles/test/musig/btc.musig.registration.test.ts

```diff
@@ -0,0 +1,292 @@
+		it("should perform a 2 out of 2 registration", async () => {
+			// TODO We shouldn't need to create a wallet first, but DW requires so, as the button to create musig wallet is inside an existing wallet
+			const first = await bitcoinWallet({
+				profile: profile1,
+				coinId: "BTC",
+				networkId: "btc.testnet",
+				mnemonic: testMnemonics["btc.testnet"][0],
+				account: 0,
+				// path: "m/48'/1'/0'/2'",
+			});
+
+			const signatory1 = new Signatories.Signatory(
+				new Signatories.MnemonicSignatory({
+					signingKey: testMnemonics["btc.testnet"][0],
+					address: "address", // Not needed / used
+					publicKey: "m/48'/1'/0'/2'", // TODO for now we use publicKey for passing path
+					privateKey: "privateKey", // Not needed / used
+				}),
+			);
+
+			const transactionData = {
+				signatory: signatory1,
+				data: {
+					min: 2,
+					numberOfSignatures: 2,
+					publicKeys: [],
+					derivationMethod: "nativeSegwitMusig",
+				},
+			};
+
+			// 1. Create and sign musig registration by first
+			const uuid = await first
+				.transaction()
+				// @ts-ignore TODO Not sure why it doesn't like the type as it is
+				.signMultiSignature(transactionData);
+			console.log(uuid);
+
+			// Broadcast the transaction to musig server with first's signature.
+			// All participant wallets should see when calling wallet.transaction.sync()
+			mockServerResponse("store", { id: uuid });
+			const result = await first.transaction().broadcast(uuid);
+
+			expect(result).toEqual({ accepted: [uuid], errors: {}, rejected: [] });
+
+			// Validate multi-signature registration data.
+			expect(first.transaction().canBeBroadcasted(uuid)).toBeFalse();
+			// expect(first.transaction().canBeSigned(uuid)).toBeFalse(); // TODO returning true because profiles is checking against the pub key of the originator's wallet (bip84 m/84'/1'/0') instead of the key that's actually used for musig (m/48'/1'/0'/2')
+			expect(first.transaction().transaction(uuid).fee()).toBeNaN();
+			expect(first.transaction().transaction(uuid).amount()).toBeNaN();
+			expect(first.transaction().transaction(uuid).timestamp()).toBeDefined();
+			expect(first.transaction().transaction(uuid).isMultiSignatureRegistration()).toBeTrue();
+			expect(first.transaction().transaction(uuid).usesMultiSignature()).toBeTrue();
+			expect(first.transaction().transaction(uuid).get("multiSignature")).toEqual({
+				min: 2,
+				numberOfSignatures: 2,
+				publicKeys: [
+					"Vpub5mtyU6Hx9xrx63Y3W4aGW1LuQkmwrq9xsQNgX7tDAM8DTHhE7vXMZ7Hue2FR8SMAGDW57fy76HFmN1jnckSmeX2cDMWVA1KViot6bLgJZuN",
+				],
+			});
+
+			mockServerResponse("pending", [oneSignatureNativeSegwitMusigRegistrationTx]);
+			mockServerResponse("ready", []);
+
+			await first.transaction().sync();
+
+			/*
+			 TODO None of the following make sense as they are passed a public key which is that of the creator's wallet, not the same used for the musig wallet. And that's by design. I can't see a way out of this.
```

Comment:
And finally here... They are all related to the same thing, as [explained here](https://cryptoarkproject.slack.com/archives/CJFKZSV61/p1637154606155800).

In short: in Ark, one mnemonic -> 1 wallet -> 1 public key. For the musig wallet, the public key that is part of the musig is the same as in the original wallet. So we can say the 2 wallets (original + musig one) are related.

For Btc, this is not the case. One mnemonic -> infinite wallets -> infinite public keys, and **_none of those infinite public keys_** is actually related to any musig wallet that could get created using the same mnemonic, because the derivation paths are on completely different branches.

So... having a btc wallet and from there creating a new musig wallet:
* doesn't make sense.
* profiles package makes **_wrong assumptions_** on the public key used for querying musig server and I'm not sure how to fix it / trick it / hack it, or whether it's possible to do it.

Reply:
See https://github.com/PayvoHQ/sdk/pull/252#discussion_r752841097